src/test/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutesV2Spec.scala

```diff
@@ -1,115 +1,156 @@
-      null,
-      null,
-      new UserService(directoryDAO, NoExtensions, Seq.empty, tosService),
-      new StatusService(directoryDAO, NoExtensions),
-      null,
-      petSAUser,
-      NoExtensions,
-      tosService = tosService
-    )
-    testCode(samRoutes, SARoutes)
-  }
+import org.broadinstitute.dsde.workbench.sam.model.api.{SamUser, SamUserAllowances, SamUserResponse}
+import org.broadinstitute.dsde.workbench.sam.service._
+import org.broadinstitute.dsde.workbench.sam.{Generator, TestSupport}
+import org.mockito.scalatest.MockitoSugar
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
 
-  "POST /register/user/v2/self" should "create user" in withDefaultRoutes { samRoutes =>
-    Post("/register/user/v2/self") ~> samRoutes.route ~> check {
-      status shouldEqual StatusCodes.Created
-      val res = responseAs[UserStatus]
-      res.userInfo.userSubjectId.value.length shouldBe 21
-      res.userInfo.userEmail shouldBe defaultUserEmail
-      res.enabled shouldBe TestSupport.enabledMapNoTosAccepted
-    }
+class UserRoutesV2Spec extends AnyFlatSpec with Matchers with ScalatestRouteTest with MockitoSugar with TestSupport {
+  val defaultUser: SamUser = Generator.genWorkbenchUserGoogle.sample.get
+  val otherUser: SamUser = Generator.genWorkbenchUserGoogle.sample.get
+  val adminGroupEmail: WorkbenchEmail = Generator.genFirecloudEmail.sample.get
+  val allUsersGroup: BasicWorkbenchGroup = BasicWorkbenchGroup(CloudExtensions.allUsersGroupName, Set(), WorkbenchEmail("all_users@fake.com"))
+
+  "GET /api/users/v2/self" should "get the user object of the requesting user" in {
+    // Arrange
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUser(defaultUser) // "persisted/enabled" user we will check the status of
+      .withAllowedUser(defaultUser) // "allowed" user we will check the status of
+      .callAsNonAdminUser()
+      .build
 
-    Post("/register/user/v2/self") ~> samRoutes.route ~> check {
-      status shouldEqual StatusCodes.Conflict
+    // Act and Assert
+    Get(s"/api/users/v2/self") ~> samRoutes.route ~> check {
+      status shouldEqual StatusCodes.OK
+      responseAs[SamUserResponse] should beForUser(defaultUser)
     }
   }
 
-  "GET /register/user/v2/self/info" should "get the status of an enabled user" in withDefaultRoutes { samRoutes =>
-    Get("/register/user/v2/self/info") ~> samRoutes.route ~> check {
-      status shouldEqual StatusCodes.NotFound
-    }
-    val (user, samDep, routes) = createTestUser(tosAccepted = true)
-    Get("/register/user/v2/self/info") ~> routes.route ~> check {
+  "GET /api/users/v2/{sam_user_id}" should "return the regular user if they're getting themselves" in {
+    // Arrange
+    val samRoutes = new MockSamRoutesBuilder(allUsersGroup)
+      .withEnabledUsers(Seq(defaultUser, otherUser))
+      .withAllowedUsers(Seq(defaultUser, otherUser))
+      .callAsNonAdminUser()
+      .build
```

Comment:
I think now that we have logic dependant on if the calling user's idenitity is an admin or not, we are going to need to pass args into `callAsNonAdminUser` since it would not be true for any arbitrary non admin, it is only true of the nonAdmin who is calling is in fact the user that is being gotten

Reply:
Since we're testing the logic based on the calling user, not the requested user, I think the functionality is still valid. It is kinda un-intuitive though, making _all_ calls to the routes be as an Admin user. I think it could be refactored, but I don't think its necessary to do in this PR

Reply:
Added the `samUser` arg to `callAs(Non)AdminUser` methods

Reply:
@tlangs you're 100% right that needing to construct the routes _WITH_ the one and only 1 user that we will be calling those routes with is indeed gross.  It's a function of how the routes are written and where we do the permissions checks (pro-tip:  don't sprinkle business logic like authorization checks in your routing logic).  This is definitely a refactoring we should do at some point, and for now, the `callAsFooUser` was the best solution I could come up with that read somewhat nicely and got us around the yucky.